### PR TITLE
Create release leader documentation

### PR DESCRIPTION
**Pull request by jdumars**
As of release 1.10, this is the most current view of the release process.

### CODE REVIEW

File: release-process-documentation/release-team-guides/release-lead.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+* This role requires a tremendous amount of experience with the Kubernetes community, code layout, ecosystem projects, organizational norms, governance, SIG structure, architecture, and release process
+
+* To properly serve as a release lead, you must have the following Kubernetes-specific qualifications:
+
+    * If you have never held a release role before, you must shadow the release lead for two full release cycles
+
+    * If you have held a release role, you must shadow the release lead for one full release cycle
+
+    * You must be a member in good standing, and part of the Kubernetes org
```
Comment (idvoretskyi) [inline release-process-documentation/release-team-guides/release-lead.md]: > You must be a member in good standing

Can you formally define this requirement?

File: release-process-documentation/release-team-guides/release-lead.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    * You must be active in at least one Kubernetes SIG
+
+    * You must be a member of the kubernetes-dev mailing list
+
+* Several release team members are in the Pacific Time Zone partially due to infrastructure access limitations at Google, therefore, the release lead must be comfortable scheduling meetings and events compatible with PT working hours
```
Comment (idvoretskyi) [inline release-process-documentation/release-team-guides/release-lead.md]: I'd rephrase this a bit. Release lead should ensure that the scheduled meetings are convenient for every release team member, keeping in mind that they can be located anywhere in the world (but with PT as a de-facto default timezone).

File: release-process-documentation/release-team-guides/release-lead.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+* Project management experience is highly desirable
+
+* Strong written and verbal communications skills are required
+
+* As the public face of the community during release time, you must do so with a very high level of professionalism
```
Comment (idvoretskyi) [inline release-process-documentation/release-team-guides/release-lead.md]: during and after :)

a lot of public conversations are happening after the release cycle

File: release-process-documentation/release-team-guides/release-lead.md (hunk 1 above)
Comment (jdumars) [reply release-process-documentation/release-team-guides/release-lead.md]: Wasn't this defined as part of the voting for steering committee nominations?  I was under the impression that there's a canonical list somewhere.

File: release-process-documentation/release-team-guides/release-lead.md (hunk 2 above)
Comment (jdumars) [reply release-process-documentation/release-team-guides/release-lead.md]: Great suggestion.  I will change that wording.

File: release-process-documentation/release-team-guides/release-lead.md (hunk 3 above)
Comment (jdumars) [reply release-process-documentation/release-team-guides/release-lead.md]: +1

File: release-process-documentation/release-team-guides/release-lead.md (hunk 1 above)
Comment (idvoretskyi) [reply release-process-documentation/release-team-guides/release-lead.md]: Alright, that's fine for now. Needs to be updated in the future is it won't be clear for someone.

**Comment by idvoretskyi** [comment]
/lgtm for now

Suggestions are minor and can be incorporated in the further versions.
